doc/src/modules/polys/basics.rst

```diff
@@ -14,6 +14,387 @@ polynomials within SymPy. All code examples assume::
+with, or *associates* of, each other. The associates of an integer
+`n` are `n` and `-n`. In a polynomial ring over a field the associates
+of a polynomial are its constant multiples.
+
+Each element of `P` is divisible by its associates and the units. An
+element is *irreducible* if it has no other divisors and is not a unit.
+The irreducible elements in the ring of integers are the prime numbers
+`p` and their opposites `-p`. In a field, every nonzero element is
+invertible and there are no irreducible elements.
+
+Factorial domains
+`````````````````
+In the ring of integers, each nonzero element can be represented as
+a product irreducible elements and optionally a unit `\pm 1`. Moreover,
+any two such products have the same number of irreducible factors
+which are associated with each other in a suitable order.
+Integral domains having this property are called *factorial*,
+or *unique factorization domains*. In addition to the ring of
+integers, all polynomial rings over a field are factorial, and so are
+more generally polynomial rings over any factorial domain. Fields
+are trivially factorial since there are only units. The irreducible
+elements of a factorial domain are usually called *primes*.
+
+A family of integers has only a finite number of common divisors
+and the greatest of them is divisible by all of them. More generally,
+given a family of nonzero elements `(a_i)` in an integral domain,
+a common divisor `d` of the elements is is called a *greatest common
+divisor*, abbreviated *gcd*, of the the family if it is a multiple
+of all common divisors.
+A greatest common divisor, if it exists, is not unique in general;
+all of its associates have the same property. It is denoted by
+`d = \gcd(a_1,\ldots,a_n)` if there is no danger of confusion.
+A *least common multiple*, or *lcm*, of a family `(a_i)` is defined
+analogously as a common multiple `m` that divides all common multiples.
+It is denoted by `m = \operatorname{lcm}(a_1,\dots,a_n)`.
+
+In a factorial domain, greatest common divisors always exists. They can
+be found, at least in principle, by factoring each element of a family
+into a product of prime powers and an optional unit, and, for each prime,
+taking the least power that appears in the factorizations.
+The product of these prime powers is then a greatest common divisor.
+A least common multiple can be obtained from the same factorizations
+as the product of the greatest powers for each prime.
+
+Euclidean domains
+`````````````````
+A practical algorithm for computing a greatest common divisor can
+be implemented in *Euclidean domains*. They are integral domains that
+can be endowed with a function `w` assigning a nonnegative integer to
+each nonzero element of the domain and having the following property:
+
+    if `a` and `b` are nonzero, there are `q` and `r` that satisfy
+    the *division identity*
+
+        `a = qb + r`
+
+    and such that either `r = 0` or `w(r) < w(b)`.
```

Comment:
you could add 'are' after 'and' if you want to keep the 'and', but I don't see a problem with just writing

"such that either `r = 0` or `w(r) < w(b)`. "

But since you don't refer to 'q' you might just say

"where either `r = 0` or `w(r) < w(b)`. "